src/plugins/data/server/index_patterns/routes/create_index_pattern.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { IndexPatternSpec } from 'src/plugins/data/common';
+import { IRouter } from '../../../../../core/server';
+import { assertIndexPatternsContext } from './util/assert_index_patterns_context';
+import { handleErrors } from './util/handle_errors';
+import { fieldSpecSchema, serializedFieldFormatSchema } from './util/schemas';
+
+const indexPatternSpecSchema = schema.object({
+  title: schema.string(),
+
+  id: schema.maybe(schema.string()),
+  version: schema.maybe(schema.string()),
+  type: schema.maybe(schema.string()),
+  intervalName: schema.maybe(schema.string()),
+  timeFieldName: schema.maybe(schema.string()),
+  sourceFilters: schema.maybe(
+    schema.arrayOf(
+      schema.object({
+        value: schema.string(),
+      })
+    )
+  ),
+  fields: schema.maybe(schema.recordOf(schema.string(), fieldSpecSchema)),
+  typeMeta: schema.maybe(schema.object({}, { unknowns: 'allow' })),
+  fieldFormats: schema.maybe(schema.recordOf(schema.string(), serializedFieldFormatSchema)),
+  fieldAttrs: schema.maybe(
+    schema.recordOf(
+      schema.string(),
+      schema.object({
+        customLabel: schema.maybe(schema.string()),
+        count: schema.maybe(schema.number()),
+      })
+    )
+  ),
+});
+
+export const registerCreateIndexPatternRoute = (router: IRouter) => {
+  router.post(
+    {
+      path: '/api/index_patterns/index_pattern',
+      validate: {
+        body: schema.object({
+          override: schema.maybe(schema.boolean({ defaultValue: false })),
+          refresh_fields: schema.maybe(schema.boolean({ defaultValue: true })),
```

Comment:
lets default to false. the reason to refresh the field list would be because you're working with the returned index pattern object. I'm not aware of any cases where that would be useful

Reply:
Updated. Good catch! Forgot to set it to `false`.